src/common/bots/Bot.cpp

```diff
@@ -6,10 +6,23 @@ Bot::Bot(){};
  * Once it has enough of a resource, it will move to the next closest mine with a resource it needs.
  */
 std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
-Bot::getNearestGreedyMove(std::vector<std::shared_ptr<Mine>> mines,
-                          std::array<uint32_t, RESOURCE_TYPE_COUNT> current_resources,
-                          sf::Vector2f bot_center,
-                          std::unordered_map<uint32_t, uint32_t> mine_id_to_resource_count) {
+Bot::getNearestGreedyMove(uint32_t bot_player_id, GameObjectController& goc) {
+    auto gso = goc.getGameStateObject();
+    auto bot_center = goc.getPlayerPosition(bot_player_id);
+    auto resource_controller = goc.getResourceController();
+    auto& player_controller = goc.getPlayerController();
+    auto current_resources = resource_controller.get(bot_player_id);
+    auto mines = (goc.getMineController()).getMines();
```

Comment:
call `goc.getMines()` instead

Reply:
it's in GameObjectStore not in the controller